### PR TITLE
fix: Add INPROGRESS artifact status update (ENG-4162)

### PR DESCRIPTION
**Pull request by mukundbayyaram**
## Conventional Commit
fix: Add INPROGRESS artifact status update [ENG-4162](https://cloudelements.atlassian.net/browse/ENG-4162)

### CODE REVIEW

File: src/core/elements/getElements.js [hunk 1]
```diff
@@ -31,17 +31,29 @@ const downloadElements = async (elements, query, jobId, processId, isPrivate, jo
...
-      logDebug(`Downloading element for element key - ${element.key}`, jobId);
-      const exportedElement = await http.get(makePath(element), query, account);
-      logDebug(`Downloaded element for element key - ${element.key}`, jobId);
-
       emitter.emit(EventTopic.ASSET_STATUS, {
         processId,
         assetType: Assets.ELEMENTS,
         assetName: element.key,
-        assetStatus: equals(jobType, JobType.PROMOTE_EXPORT) ? ArtifactStatus.INPROGRESS : ArtifactStatus.COMPLETED,
+        assetStatus: ArtifactStatus.INPROGRESS,
```
Comment (MohdRafiUddin) [inline src/core/elements/getElements.js]: Do the same for upload as well

File: src/core/elements/getElements.js [hunk 2]
```diff
@@ -31,17 +31,29 @@ const downloadElements = async (elements, query, jobId, processId, isPrivate, jo
...
+      logDebug(`Downloaded element for element key - ${element.key}`, jobId);
+
+      // If 'promote_export' job, the final artifact
+      // status update will happen in doctor-service
+      if (!equals(jobType, JobType.PROMOTE_EXPORT)) {
+        emitter.emit(EventTopic.ASSET_STATUS, {
+          processId,
+          assetType: Assets.ELEMENTS,
+          assetName: element.key,
+          assetStatus: ArtifactStatus.COMPLETED,
```
Comment (MohdRafiUddin) [inline src/core/elements/getElements.js:55]: we need the promote condition here

**Review by MohdRafiUddin — CHANGES_REQUESTED** [review]

File: src/core/elements/getElements.js (hunk 2 above)
Comment (mukundbayyaram) [reply src/core/elements/getElements.js:55]: The promote condition is there in line 48.
If it's a promote job, we simply don't update the status to `completed`

File: src/core/elements/getElements.js (hunk 1 above)
Comment (mukundbayyaram) [reply src/core/elements/getElements.js]: ~~Regardless of the jobType, we have to update the status to `INPROGRESS`.~~
**EDIT:** Done

**Review by MohdRafiUddin — APPROVED** [review]

**Pull request merged**